docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -1,466 +1,1741 @@
+Preparing the Database
+~~~~~~~~~~~~~~~~~~~~~~
 
-PQS Schema Design
-=================
+Start by investigating the database schema that PQS will create, using
+the ``datastore`` command that is used for administrative operations:
 
-PQS is not directly involved in querying/reading the datastore - the
-application is free to query it, such as via JDBC. The objectives of the
-schema design is to facilitate:
+.. code:: sql
 
--  **Scalable writes**: transactions are written in parallel, so
-   writes do not need to be sequential.
--  **Scalable reads**: queries can be parallelized and are not
-   blocked by writes. They produce sensible query plans with no
-   unnecessary table scans.
--  **Ease of use**: readers can use familiar tools and techniques to
-   query the datastore without needing to understand the specifics of
-   the schema design. Simple entry points
-   provide access to data in familiar ways. In particular, readers
-   do not need to navigate the offset-based model.
--  **Read consistency**: readers can achieve the level of
-   consistency that they require, including consistency with other
-   ledger datastores, or with ledger commands that have been executed.
-
-The following principles apply:
-
--  **Append-only**: only INSERTs are used, and no UPDATEs or DELETEs are
-   used in transaction processing.
--  **Offset-based**: all physical tables are indexed by offset, meaning that
-   all ledger data is known in terms of the offset in which it was
-   committed to the ledger.
--  **Implicit offset**: readers can opt for queries with implicit offset,
-   meaning they can ignore the role of offset in their queries but
-   still provide a stable view of the ledger data. Much like PostgreSQL
-   provides MVCC capabilities without the reader needing to understand
-   the underlying implementation, this provides a similar
-   experience for readers of the ledger data.
--  **Idempotent**: PQS is designed to be restarted at any time, and it will
-   not impact the integrity of the data. This is achieved by using the
-   offset-based model and ensuring that (other than the datastore
-   itself) PQS is stateless.
--  **Watermarks**: PQS maintains a watermark of the latest contiguous
-   offset, representing the point of the ledger that has been fully
-   processed. This is used to ensure that the ledger data has read
-   consistency, without requiring readers to perform table
-   scans. This resolves the uncertainty created by the
-   parallel writes.
-
-JSON Data
-=========
+   $ ./scribe.jar datastore postgres-document schema show
+   [...]
+   /**********************************************************
+    * generated by scribe, version: [...]                    *
+    **********************************************************/
+   -- tables
+   create table [...]
+
+With the produced DDL you can create the schema yourself - or leave it
```

Comment:
Under what circumstances would we want customers to use the DDL to create the schema?  I think they could inspect it but I'm not sure we want any customers fingers to have to be involved in setting things up because they will probably mess it up.

Reply:
A DBA worth their salt will want to create the db themselves, and apply the schema manually after inspecting - and not trust a tool like Scribe to do this invisibly for them. Then they know what it is doing to their db server, eg. not installing any extensions, weird collations or other server settings they are not happy with. The intent is not that they edit the schema - but inspect & apply.